docs/reference/job-schema.rst

```diff
@@ -0,0 +1,61 @@
+Test job schema
+=================
+
+Test jobs can be defined in either YAML or JSON format.
+
+The following table lists the key elements that a job definition file should contain.
+
+.. list-table:: Test job schema
+  :header-rows: 1
+
+  * - Field
+    - Type 
+    - Default
+    - Description
+  * - ``job_queue``
+    - String
+    - /
+    - Name of the job queue to which you want to submit the job. This field is mandatory.
+  * - ``global_timeout``
+    - integer
+    - | 14400
+      | (4 hours)
+    - | (Optional) Maximum time (in seconds) a job is allowed to run. If the global timeout is reached while a job is still running, Testflinger will close the job immediately. 
+      | You can set a global timeout larger than the default value, but the per-device configuration might have a more restrict timeout and overwrite the global value.
+  * - ``output_timeout``
+    - integer
+    - | 900
+      | (15 minutes)
+    - (Optional) Maximum time (in seconds) Testflinger should wait in the ``test`` phase to get output from the job. If the timeout is reached before the test phase has any output, Testflinger will cancel the job. This value should be smaller than, or equal to, the ``global_timeout``. 
+  * - ``allocation_timeout``
+    - integer
+    - | 7200
+      | (2 hours)
+    - (Optional) Maximum time (in seconds) Testflinger should wait in the ``allocate`` phase for multi-device jobs to reach the ``allocated`` state. If the timeout is reached before all devices are allocated, Testflinger will cancel the job.
+  * - ``<phase>_data``
+    - dictionary
+    - /
+    - | (Optional) Sections that define the configurations and commands about how the job should be executed in each of the test phases. If any phase encounters a non-zero exit code, no further phases will be executed.
+      | Supported test phases include: 
+      |   - provision
+      |   - firmware_update
+      |   - test
+      |   - allocate
+      |   - reserve 
+      | For detailed information about how to define the data to include in each test phase, see :doc:`test-phases`. 
+
+Example job in YAML
+----------------------------
+
+The following example YAML file defines a job that retrieves the Ubuntu 22.04-jammy  system image from the given URL, provisions the image on the device at IP address ``DEVICE_IP``, and prints the distribution-specific information on the provisioned device:
+
+.. code-block:: yaml
+
+  job-queue: example-queue
+  global-timeout: 28800
+  output-timeout: 3600
+  provision_data:
+    url: https://cdimage.ubuntu.com/ubuntu/releases/jammy/release/example.img.xz
```

Comment:
iirc, this example use to show the use of a maas system. If you are are going to use a `url: ...` tag in the provision data, then I think we should be really careful to make it clear that this is a muxpi provisioned device like rpi or risc-v. For instance, maybe the queue name could be `rpi4b` and the url could actually point to the arm64-raspi image for jammy?

If you are trying to show a maas image, then the distro tag needs to be used. MaaS provisioned devices cannot specify a url, so we should not give the impression that this will work for all types of systems.

Reply:
I think that might be useful, yes. Maybe one with `job_queue: example-maas-queue` and another with `job_queue: rpi4b` or something? That might make it more obvious. You could even use something like a comment in the yaml `# this is a jammy server image` or something to make it clear and hopefully handle Kevin's concern.

Reply:
Oh... and I just saw your updates after typing that ^
So I think you anticipated almost everything I was going to say, only one minor change to that rpi4b job queue name, but otherwise looks good to me!